tests/topotests/lib/pim.py

```diff
@@ -1935,36 +2209,105 @@ def clear_pim_interfaces(tgen, dut):
 
     logger.debug("Entering lib API: {}".format(sys._getframe().f_code.co_name))
 
-    nh_before_clear = {}
-    nh_after_clear = {}
+    for dut in tgen.routers():
+        if "pim" not in topo["routers"][dut]:
+            continue
 
-    rnode = tgen.routers()[dut]
+        rnode = tgen.routers()[dut]
 
-    logger.info("[DUT: %s]: Verify pim neighbor before pim" " neighbor clear", dut)
-    # To add uptime initially
-    sleep(10)
-    run_json_before = run_frr_cmd(rnode, "show ip pim neighbor json", isjson=True)
+        logger.info("[DUT: %s]: Clearing pim6 interface traffic", dut)
+        result = run_frr_cmd(rnode, "clear ipv6 pim interface traffic")
 
-    for key, value in run_json_before.items():
-        if bool(value):
-            for _key, _value in value.items():
-                nh_before_clear[key] = _value["upTime"]
+    logger.debug("Exiting lib API: {}".format(sys._getframe().f_code.co_name))
 
-    # Clearing PIM neighbors
-    logger.info("[DUT: %s]: Clearing pim interfaces", dut)
-    run_frr_cmd(rnode, "clear ip pim interfaces")
+    return True
 
-    logger.info("[DUT: %s]: Verify pim neighbor after pim" " neighbor clear", dut)
 
-    found = False
+def clear_pim6_interfaces(tgen, topo):
+    """
+    Clear ipv6 pim interface by running
+    "clear ipv6 pim interface" cli
 
-    # Waiting for maximum 60 sec
-    fail_intf = []
-    for retry in range(1, 13):
-        logger.info("[DUT: %s]: Waiting for 5 sec for PIM neighbors" " to come up", dut)
-        sleep(5)
-        run_json_after = run_frr_cmd(rnode, "show ip pim neighbor json", isjson=True)
-        found = True
+    Parameters
+    ----------
+    * `tgen`: topogen object
+    Usage
+    -----
+
+    result = clear_pim6_interfaces(tgen, topo)
+
+    Returns
+    -------
+    errormsg(str) or True
+    """
+
+    logger.debug("Entering lib API: {}".format(sys._getframe().f_code.co_name))
+
+    for dut in tgen.routers():
+        if "pim" not in topo["routers"][dut]:
+            continue
+
+        rnode = tgen.routers()[dut]
+
+        logger.info("[DUT: %s]: Clearing pim6 interfaces", dut)
+        result = run_frr_cmd(rnode, "clear ipv6 pim interface")
+
+    logger.debug("Exiting lib API: {}".format(sys._getframe().f_code.co_name))
+
+    return True
+
+
+def clear_pim_interfaces(tgen, dut):
+    """
+    Clear ip/ipv6 pim interface by running
+    "clear ip/ipv6 pim interfaces" cli
+
+    Parameters
+    ----------
+    * `tgen`: topogen object
+    * `dut`: Device Under Test
+    Usage
+    -----
+
+    result = clear_pim_interfaces(tgen, dut)
+
+    Returns
+    -------
+    errormsg(str) or True
+    """
+
+    logger.debug("Entering lib API: {}".format(sys._getframe().f_code.co_name))
+
+    nh_before_clear = {}
+    nh_after_clear = {}
+
+    rnode = tgen.routers()[dut]
+
+    logger.info("[DUT: %s]: Verify pim neighbor before pim" " neighbor clear", dut)
+    # To add uptime initially
+    sleep(10)
```

Comment:
Can we get rid of sleep and replace with run_and_expect?

Reply:
Done